package com.igexin.push.core.a.c;

import android.app.NotificationManager;
import android.content.Context;
import android.database.Cursor;
import android.text.TextUtils;
import com.igexin.assist.control.AbstractPushManager;
import com.igexin.push.core.b.t;
import com.igexin.push.core.d;
import com.igexin.push.extension.mod.BaseActionBean;
import com.igexin.push.extension.mod.PushMessageInterface;
import com.igexin.push.extension.mod.PushTaskBean;
import com.xiaomi.mipush.sdk.MiPushClient;
import java.util.HashSet;
import org.json.JSONObject;

/* JADX INFO: compiled from: SearchBox */
/* JADX INFO: loaded from: classes8.dex */
public class m implements PushMessageInterface {

    /* JADX INFO: renamed from: a, reason: collision with root package name */
    private static final String f7165a = com.igexin.push.core.b.f + m.class.getName();

    private static void a(String str) {
        try {
            com.igexin.c.a.c.a.a(f7165a + "|del condition taskid = " + str, new Object[0]);
            d.a.f7200a.i.a("message", new String[]{"taskid"}, new String[]{str});
        } catch (Throwable th) {
            com.igexin.c.a.c.a.a(th);
            com.igexin.c.a.c.a.b(f7165a, "del condition" + th.toString());
        }
    }

    /* JADX WARN: Removed duplicated region for block: B:100:0x017e A[EXC_TOP_SPLITTER, SYNTHETIC] */
    /* JADX WARN: Removed duplicated region for block: B:54:0x0163  */
    /* JADX WARN: Removed duplicated region for block: B:56:0x0166 A[PHI: r4 r5
      0x0166: PHI (r4v3 boolean) = (r4v2 boolean), (r4v10 boolean) binds: [B:61:0x0170, B:55:0x0164] A[DONT_GENERATE, DONT_INLINE]
      0x0166: PHI (r5v4 android.database.Cursor) = (r5v3 android.database.Cursor), (r5v6 android.database.Cursor) binds: [B:61:0x0170, B:55:0x0164] A[DONT_GENERATE, DONT_INLINE]] */
    /* JADX WARN: Removed duplicated region for block: B:98:0x024f  */
    @Override // com.igexin.push.extension.mod.PushMessageInterface
    /*
        Code decompiled incorrectly, please refer to instructions dump.
    */
    public boolean executeAction(PushTaskBean pushTaskBean, BaseActionBean baseActionBean) {
        Cursor cursorA;
        boolean z;
        String strA;
        PushTaskBean pushTaskBean2;
        String str;
        t tVar = (t) baseActionBean;
        String str2 = tVar.f7188a;
        NotificationManager notificationManager = (NotificationManager) com.igexin.push.core.e.l.getSystemService("notification");
        if (TextUtils.isEmpty(str2)) {
            z = false;
        } else {
            try {
                cursorA = d.a.f7200a.i.a("message", new String[]{"taskid"}, new String[]{str2}, null, "id ASC");
            } catch (Throwable th) {
                th = th;
                cursorA = null;
            }
            if (cursorA != null) {
                try {
                    if (cursorA.moveToFirst()) {
                        String string = cursorA.getString(cursorA.getColumnIndex("messageid"));
                        com.igexin.push.core.a.b.d();
                        strA = com.igexin.push.core.a.b.a(str2, string);
                    } else {
                        strA = "";
                    }
                    if (strA.equals("") || (pushTaskBean2 = com.igexin.push.core.e.ah.get(strA)) == null) {
                        z = false;
                    } else {
                        pushTaskBean2.setStop(true);
                        try {
                            com.igexin.c.a.c.a.a(f7165a + "|del condition taskid = " + str2, new Object[0]);
                            d.a.f7200a.i.a("message", new String[]{"taskid"}, new String[]{str2});
                        } catch (Throwable th2) {
                            com.igexin.c.a.c.a.a(th2);
                            com.igexin.c.a.c.a.b(f7165a, "del condition" + th2.toString());
                        }
                        String strValueOf = String.valueOf(pushTaskBean2.getPerActionid());
                        if (!strValueOf.equals("0") && pushTaskBean2.getBaseAction(strValueOf).getType().equals("notification") && com.igexin.push.core.e.ai.containsKey(str2)) {
                            notificationManager.cancel(com.igexin.push.core.e.ai.get(str2).intValue());
                            com.igexin.push.core.e.ai.remove(str2);
                            try {
                                BaseActionBean actionByType = pushTaskBean2.getActionByType("notification");
                                if (actionByType != null) {
                                    com.igexin.push.core.b.l lVar = (com.igexin.push.core.b.l) actionByType;
                                    String str3 = lVar.q;
                                    HashSet<String> hashSet = com.igexin.push.core.e.aj.get(str3);
                                    Integer num = com.igexin.push.core.e.ak.get(lVar.q);
                                    if (hashSet != null && !hashSet.isEmpty()) {
                                        hashSet.remove(str2);
                                    }
                                    if (!TextUtils.isEmpty(str3) && num != null && hashSet != null && hashSet.isEmpty()) {
                                        ((NotificationManager) com.igexin.push.core.e.l.getSystemService("notification")).cancel(num.intValue());
                                        com.igexin.push.core.e.aj.remove(str3);
                                        com.igexin.push.core.e.ak.remove(str3);
                                    }
                                }
                                z = true;
                            } catch (Throwable th3) {
                                th = th3;
                                z = true;
                                try {
                                    com.igexin.c.a.c.a.a(th);
                                    if (cursorA != null) {
                                    }
                                    if (!z) {
                                    }
                                    if (!baseActionBean.getDoActionId().equals("")) {
                                    }
                                    return true;
                                } finally {
                                }
                            }
                        } else {
                            z = false;
                        }
                        try {
                            com.igexin.push.core.i.a aVarA = com.igexin.push.core.i.b.a().a(Long.valueOf(com.igexin.push.core.g.b));
                            if (aVarA != null && aVarA.b().equals(str2)) {
                                com.igexin.push.core.i.b.a().a(aVarA);
                            }
                        } catch (Throwable th4) {
                            th = th4;
                            com.igexin.c.a.c.a.a(th);
                            if (cursorA != null) {
                            }
                        }
                    }
                } catch (Throwable th5) {
                    th = th5;
                    z = false;
                    com.igexin.c.a.c.a.a(th);
                    if (cursorA != null) {
                    }
                    if (!z) {
                    }
                    if (!baseActionBean.getDoActionId().equals("")) {
                    }
                    return true;
                }
                if (cursorA != null) {
                    cursorA.close();
                }
            }
        }
        if (!z) {
            try {
            } catch (Throwable th6) {
                com.igexin.c.a.c.a.a(th6);
            }
            if (tVar.b) {
                com.igexin.c.a.c.a.a(f7165a + " | cancelAll()", new Object[0]);
                notificationManager.cancelAll();
                com.igexin.assist.sdk.a aVarA2 = com.igexin.assist.sdk.a.a();
                Context context = com.igexin.push.core.e.l;
                AbstractPushManager abstractPushManager = aVarA2.b;
                if (abstractPushManager != null && abstractPushManager.isSupport()) {
                    if (aVarA2.b.getBrandCode().equals("3")) {
                        try {
                            String str4 = MiPushClient.COMMAND_REGISTER;
                            MiPushClient.class.getDeclaredMethod("clearNotification", Context.class).invoke(null, context);
                        } catch (Throwable th7) {
                            com.igexin.c.a.c.a.a(th7);
                            com.igexin.c.a.c.a.a("AssistMangerFactory | cancelAllAssistNotification() err " + th7.toString(), new Object[0]);
                        }
                        str = " cancelAllAssistNotification() XM ";
                    } else {
                        if (aVarA2.b.getBrandCode().equals("4")) {
                            try {
                                Class.forName("com.meizu.cloud.pushsdk.PushManager").getDeclaredMethod("clearNotification", Context.class).invoke(null, context);
                            } catch (Throwable th8) {
                                com.igexin.c.a.c.a.a(th8);
                                com.igexin.c.a.c.a.a("AssistMangerFactory | cancelAllAssistNotification() err " + th8.toString(), new Object[0]);
                            }
                            str = " cancelAllAssistNotification() MZ ";
                        }
                        com.igexin.c.a.c.a.a(th6);
                    }
                    com.igexin.c.a.c.a.b(com.igexin.assist.sdk.a.f7005a, str);
                }
                com.igexin.push.core.e.ak.clear();
                com.igexin.push.core.e.aj.clear();
            }
        }
        if (!baseActionBean.getDoActionId().equals("")) {
            com.igexin.push.core.a.b.d();
            com.igexin.push.core.a.b.a(pushTaskBean.getTaskId(), pushTaskBean.getMessageId(), baseActionBean.getDoActionId());
        }
        return true;
    }

    @Override // com.igexin.push.extension.mod.PushMessageInterface
    public BaseActionBean parseAction(JSONObject jSONObject) {
        try {
            if (!jSONObject.has("do") || !jSONObject.has("actionid") || !jSONObject.has("taskid")) {
                return null;
            }
            t tVar = new t();
            tVar.setType(com.igexin.push.core.b.o);
            tVar.setActionId(jSONObject.getString("actionid"));
            tVar.setDoActionId(jSONObject.getString("do"));
            tVar.f7188a = jSONObject.getString("taskid");
            tVar.b = jSONObject.optBoolean("force");
            return tVar;
        } catch (Exception e) {
            com.igexin.c.a.c.a.a(e);
            return null;
        }
    }

    @Override // com.igexin.push.extension.mod.PushMessageInterface
    public PushMessageInterface.ActionPrepareState prepareExecuteAction(PushTaskBean pushTaskBean, BaseActionBean baseActionBean) {
        return PushMessageInterface.ActionPrepareState.success;
    }

    private static void a(String str, BaseActionBean baseActionBean) {
        if (baseActionBean == null) {
            return;
        }
        com.igexin.push.core.b.l lVar = (com.igexin.push.core.b.l) baseActionBean;
        String str2 = lVar.q;
        HashSet<String> hashSet = com.igexin.push.core.e.aj.get(str2);
        Integer num = com.igexin.push.core.e.ak.get(lVar.q);
        if (hashSet != null && !hashSet.isEmpty()) {
            hashSet.remove(str);
        }
        if (TextUtils.isEmpty(str2) || num == null || hashSet == null || !hashSet.isEmpty()) {
            return;
        }
        ((NotificationManager) com.igexin.push.core.e.l.getSystemService("notification")).cancel(num.intValue());
        com.igexin.push.core.e.aj.remove(str2);
        com.igexin.push.core.e.ak.remove(str2);
    }
}
